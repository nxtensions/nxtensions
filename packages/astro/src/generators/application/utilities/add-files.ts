import {
  generateFiles,
  joinPathFragments,
  names,
  offsetFromRoot,
  Tree,
} from '@nrwl/devkit';
import { NormalizedGeneratorOptions } from '../schema';

export function addFiles(
  tree: Tree,
  options: NormalizedGeneratorOptions
): void {
  generateFiles(
    tree,
    joinPathFragments(__dirname, '..', 'files', 'project'),
    options.projectRoot,
    {
      ...options,
      ...names(options.name),
      offsetFromRoot: offsetFromRoot(options.projectRoot),
      tmpl: '',
    }
  );

  if (options.integrations.find((i) => i.name === 'tailwind')) {
    generateFiles(
      tree,
      joinPathFragments(__dirname, '..', 'files', 'tailwind'),
      options.projectRoot,
      { tmpl: '' }
    );
  }
}
