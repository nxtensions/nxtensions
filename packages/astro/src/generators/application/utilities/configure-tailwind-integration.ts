import { joinPathFragments, readJson, type Tree } from '@nx/devkit';
import { insertImport } from '@nx/js';
import { ast, query } from '@phenomnomnominal/tsquery';
import { clean, coerce, lt } from 'semver';
import type { CallExpression } from 'typescript';
import type { IntegrationInfo, NormalizedGeneratorOptions } from '../schema';

export function configureTailwindIntegration(
  tree: Tree,
  options: NormalizedGeneratorOptions
): void {
  const tailwindIntegration = options.integrations.find(
    (integration) => integration.name === 'tailwind'
  );
  if (!tailwindIntegration) {
    return;
  }

  const tailwindIntegrationVersion = getTailwindIntegrationVersion(
    tree,
    tailwindIntegration
  );
  if (lt(tailwindIntegrationVersion, '4.0.0')) {
    return;
  }

  const astroConfigPath = joinPathFragments(
    options.projectRoot,
    'astro.config.mjs'
  );
  const astroConfigContent = tree.read(astroConfigPath, 'utf-8');
  let source = ast(astroConfigContent);

  const tailwindCallExpressions = query<CallExpression>(
    source,
    'ExportAssignment CallExpression:has(Identifier[name=defineConfig]) ObjectLiteralExpression PropertyAssignment:has(Identifier[name=integrations]) ArrayLiteralExpression CallExpression:has(Identifier[name=tailwind])'
  );
  if (!tailwindCallExpressions.length) {
    return;
  }

  const index = tailwindCallExpressions[0].getEnd() - 1;
  const updatedConfig =
    astroConfigContent.slice(0, index) +
    `{
      configFile: fileURLToPath(
        new URL('./tailwind.config.cjs', import.meta.url)
      ),
    }` +
    astroConfigContent.slice(index);

  tree.write(astroConfigPath, updatedConfig);

  // add import for `fileURLToPath`
  source = ast(updatedConfig);
  insertImport(tree, source, astroConfigPath, 'fileURLToPath', 'node:url');
}

function getTailwindIntegrationVersion(
  tree: Tree,
  integration: IntegrationInfo
): string | null {
  const { dependencies, devDependencies } = readJson(tree, 'package.json');
  const installedTailwindVersion =
    dependencies?.[integration.packageName] ??
    devDependencies?.[integration.packageName];

  if (installedTailwindVersion) {
    return cleanSemver(installedTailwindVersion);
  }

  const [, integrationVersion] = integration.dependencies.find(
    ([name]) => name === integration.packageName
  );

  return cleanSemver(integrationVersion);
}

function cleanSemver(version: string): string {
  return clean(version) ?? coerce(version).version;
}
