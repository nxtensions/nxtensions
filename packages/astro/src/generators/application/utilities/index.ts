export * from './add-files';
export * from './add-project';
export * from './install-renderers';
export * from './normalize-options';
export * from './setup-e2e-tests';
