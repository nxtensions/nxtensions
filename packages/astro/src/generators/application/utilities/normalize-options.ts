import { getWorkspaceLayout, names, Tree } from '@nrwl/devkit';
import { GeneratorOptions, NormalizedGeneratorOptions } from '../schema';

export function normalizeOptions(
  tree: Tree,
  options: GeneratorOptions
): NormalizedGeneratorOptions {
  const { appsDir, standaloneAsDefault } = getWorkspaceLayout(tree);

  const name = names(options.name).fileName;
  const directory = options.directory
    ? `${names(options.directory).fileName}/${name}`
    : name;
  const projectName = directory.replace(/\//g, '-');
  const projectRoot = `${appsDir}/${directory}`;
  const tags = options.tags
    ? options.tags.split(',').map((tag) => tag.trim())
    : [];

  return {
    ...options,
    addCypressTests: options.addCypressTests ?? true,
    projectName,
    projectRoot,
    renderers: options.renderers ?? [],
    standaloneConfig: options.standaloneConfig ?? standaloneAsDefault,
    tags,
  };
}
