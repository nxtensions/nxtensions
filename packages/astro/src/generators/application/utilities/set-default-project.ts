import { readNxJson, Tree, updateNxJson } from '@nx/devkit';

export function setDefaultProject(tree: Tree, projectName: string): void {
  const workspace = readNxJson(tree);
  if (!workspace.defaultProject) {
    workspace.defaultProject = projectName;
    updateNxJson(tree, workspace);
  }
}
