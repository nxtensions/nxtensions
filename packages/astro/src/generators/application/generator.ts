import {
  ensurePackage,
  formatFiles,
  type GeneratorCallback,
  type Tree,
} from '@nx/devkit';
import { addDependenciesToPackageJson } from '../../utilities/package-json';
import { initGenerator } from '../init/generator';
import {
  astroCheckVersion,
  astroVersion,
  getInstalledNxVersion,
  typescriptVersion,
} from '../utilities/versions';
import type { GeneratorOptions } from './schema';
import {
  addFiles,
  addIntegrationsPackages,
  addProject,
  configureTailwindIntegration,
  normalizeOptions,
  setDefaultProject,
  setupE2ETests,
} from './utilities';

export async function applicationGenerator(
  tree: Tree,
  rawOptions: GeneratorOptions
): Promise<GeneratorCallback | void> {
  return await applicationGeneratorInternal(tree, {
    projectNameAndRootFormat: 'derived',
    ...rawOptions,
  });
}

export async function applicationGeneratorInternal(
  tree: Tree,
  rawOptions: GeneratorOptions
): Promise<GeneratorCallback | void> {
  const options = await normalizeOptions(tree, rawOptions);

  const tasks: GeneratorCallback[] = [];
  const { initGenerator: jsInitGenerator } = ensurePackage<
    typeof import('@nx/js')
  >('@nx/js', getInstalledNxVersion(tree));
  const jsTask = await jsInitGenerator(tree, { skipFormat: true });
  tasks.push(jsTask);

  await initGenerator(tree, {});

  const depsTask = addDependenciesToPackageJson(
    tree,
    {},
    {
      astro: astroVersion,
      '@astrojs/check': astroCheckVersion,
      typescript: typescriptVersion,
    }
  );
  tasks.push(depsTask);

  addProject(tree, options);
  setDefaultProject(tree, options.projectName);
  addFiles(tree, options);
  await configureTailwindIntegration(tree, options);

  const e2eTask = await setupE2ETests(tree, options);
  if (e2eTask) {
    tasks.push(e2eTask);
  }

  const integrationsTask = addIntegrationsPackages(tree, options.integrations);
  if (integrationsTask) {
    tasks.push(integrationsTask);
  }

  await formatFiles(tree);

  return () => tasks.forEach((task) => task());
}

export default applicationGenerator;
