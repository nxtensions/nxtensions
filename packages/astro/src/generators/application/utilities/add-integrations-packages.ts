import type { GeneratorCallback, Tree } from '@nx/devkit';
import { addDependenciesToPackageJson } from '../../../utilities/package-json';
import type { IntegrationInfo } from '../schema';

export function addIntegrationsPackages(
  tree: Tree,
  integrations: IntegrationInfo[]
): GeneratorCallback {
  if (integrations.length === 0) {
    return undefined;
  }

  const dependencies = Array.from(
    new Set(
      integrations
        .map(({ dependencies }) => dependencies)
        .flat(1)
        .sort(([a], [b]) => a.localeCompare(b))
    )
  );

  return addDependenciesToPackageJson(
    tree,
    {},
    dependencies.reduce((acc, [packageName, version]) => {
      acc[packageName] = version;
      return acc;
    }, {})
  );
}
