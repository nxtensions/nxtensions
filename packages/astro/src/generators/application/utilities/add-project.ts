import type { Tree } from '@nrwl/devkit';
import { addProjectConfiguration, joinPathFragments } from '@nrwl/devkit';
import { gte } from 'semver';
import { getInstalledAstroVersion } from '../../../utilities/versions';
import type { NormalizedGeneratorOptions } from '../schema';

export function addProject(
  tree: Tree,
  options: NormalizedGeneratorOptions
): void {
  // eslint-disable-next-line @typescript-eslint/no-var-requires
  const { version: nxVersion } = require('nx/package.json');
  const outputDirectory = gte(nxVersion, '15.0.0')
    ? joinPathFragments('{workspaceRoot}', 'dist', '{projectRoot}')
    : joinPathFragments('dist', options.projectRoot);
  const astroVersion = getInstalledAstroVersion();

  addProjectConfiguration(
    tree,
    options.projectName,
    {
      root: options.projectRoot,
      projectType: 'application',
      sourceRoot: joinPathFragments(options.projectRoot, 'src'),
      targets: {
        build: {
          outputs: [outputDirectory],
          executor: '@nxtensions/astro:build',
          options: {},
        },
        dev: {
          executor: '@nxtensions/astro:dev',
          options: {},
        },
        preview: {
          dependsOn: [
            {
              target: 'build',
              projects: 'self',
            },
          ],
          executor: '@nxtensions/astro:preview',
          options: {},
        },
        check: {
          executor: '@nxtensions/astro:check',
        },
        sync:
          !astroVersion || gte(astroVersion, '1.8.0')
            ? { executor: '@nxtensions/astro:sync' }
            : undefined,
      },
      tags: options.tags,
    },
    options.standaloneConfig
  );
}
