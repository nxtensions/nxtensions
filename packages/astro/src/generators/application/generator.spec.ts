jest.mock('node-fetch');
jest.mock('@nx/devkit', () => ({
  ...jest.requireActual('@nx/devkit'),
  formatFiles: jest.fn(),
}));

import {
  formatFiles,
  getProjects,
  readJson,
  readNxJson,
  readProjectConfiguration,
  updateNxJson,
  type Tree,
} from '@nx/devkit';
import fetch from 'node-fetch';
import { createTreeWithEmptyWorkspace } from '../utilities/testing';
import { applicationGenerator } from './generator';
import type { GeneratorOptions } from './schema';

describe('application generator', () => {
  let tree: Tree;
  const options: GeneratorOptions = {
    name: 'app1',
    projectNameAndRootFormat: 'as-provided',
  };

  beforeEach(() => {
    tree = createTreeWithEmptyWorkspace();
    jest.clearAllMocks();
  });

  test('should add project configuration', async () => {
    await applicationGenerator(tree, options);

    const config = readProjectConfiguration(tree, options.name);
    expect(config).toMatchSnapshot();
  });

  test('should set target defaults', async () => {
    await applicationGenerator(tree, options);

    const nxJson = readNxJson(tree);
    expect(nxJson.targetDefaults).toStrictEqual(
      expect.objectContaining({
        '@nxtensions/astro:build': {
          cache: true,
          inputs: ['default', '^default'],
          outputs: ['{workspaceRoot}/dist/{projectRoot}'],
        },
        '@nxtensions/astro:check': {
          cache: true,
          inputs: ['default', '^default'],
        },
        '@nxtensions/astro:preview': { dependsOn: ['build'] },
      })
    );
  });

  test('should not set the default project when it is already set', async () => {
    let config = readNxJson(tree);
    config.defaultProject = 'other-app';
    updateNxJson(tree, config);

    await applicationGenerator(tree, options);

    config = readNxJson(tree);
    expect(config.defaultProject).toBe('other-app');
  });

  test('should set the default project when it is not set', async () => {
    await applicationGenerator(tree, options);

    const config = readNxJson(tree);
    expect(config.defaultProject).toBe(options.name);
  });

  test('should generate files', async () => {
    await applicationGenerator(tree, options);

    expect(tree.exists(`${options.name}/public/favicon.svg`)).toBeTruthy();
    expect(
      tree.exists(`${options.name}/src/components/Card.astro`)
    ).toBeTruthy();
    expect(
      tree.exists(`${options.name}/src/layouts/Layout.astro`)
    ).toBeTruthy();
    expect(tree.exists(`${options.name}/src/pages/index.astro`)).toBeTruthy();
    expect(tree.exists(`${options.name}/astro.config.mjs`)).toBeTruthy();
    expect(tree.exists(`${options.name}/tsconfig.json`)).toBeTruthy();
  });

  test('should generate files in the right location in a monorepo layout', async () => {
    tree.write('apps/.gitkeep', '');

    await applicationGenerator(tree, {
      ...options,
      projectNameAndRootFormat: undefined,
    });

    expect(tree.exists(`apps/${options.name}/public/favicon.svg`)).toBeTruthy();
    expect(
      tree.exists(`apps/${options.name}/src/components/Card.astro`)
    ).toBeTruthy();
    expect(
      tree.exists(`apps/${options.name}/src/layouts/Layout.astro`)
    ).toBeTruthy();
    expect(
      tree.exists(`apps/${options.name}/src/pages/index.astro`)
    ).toBeTruthy();
    expect(tree.exists(`apps/${options.name}/astro.config.mjs`)).toBeTruthy();
    expect(tree.exists(`apps/${options.name}/tsconfig.json`)).toBeTruthy();
  });

  test('should format files', async () => {
    await applicationGenerator(tree, options);

    expect(formatFiles).toHaveBeenCalled();
  });

  describe('--directory', () => {
    test('should add project with the right name when a directory is provided', async () => {
      const directory = `some-directory/sub-directory/${options.name}`;

      await applicationGenerator(tree, { ...options, directory });

      const project = readProjectConfiguration(tree, options.name);
      expect(project).toBeTruthy();
    });

    test('should generate files in the right directory', async () => {
      const directory = `some-directory/sub-directory/${options.name}`;

      await applicationGenerator(tree, { ...options, directory });

      expect(tree.exists(`${directory}/public/favicon.svg`)).toBeTruthy();
      expect(
        tree.exists(`${directory}/src/components/Card.astro`)
      ).toBeTruthy();
      expect(tree.exists(`${directory}/src/layouts/Layout.astro`)).toBeTruthy();
      expect(tree.exists(`${directory}/src/pages/index.astro`)).toBeTruthy();
      expect(tree.exists(`${directory}/astro.config.mjs`)).toBeTruthy();
      expect(tree.exists(`${directory}/tsconfig.json`)).toBeTruthy();
    });

    test('should generate files in the right directory in a monorepo layout', async () => {
      tree.write('apps/.gitkeep', '');
      const directory = 'some-directory/sub-directory';

      await applicationGenerator(tree, {
        ...options,
        directory,
        projectNameAndRootFormat: undefined,
      });

      expect(
        tree.exists(`apps/${directory}/${options.name}/public/favicon.svg`)
      ).toBeTruthy();
      expect(
        tree.exists(
          `apps/${directory}/${options.name}/src/components/Card.astro`
        )
      ).toBeTruthy();
      expect(
        tree.exists(
          `apps/${directory}/${options.name}/src/layouts/Layout.astro`
        )
      ).toBeTruthy();
      expect(
        tree.exists(`apps/${directory}/${options.name}/src/pages/index.astro`)
      ).toBeTruthy();
      expect(
        tree.exists(`apps/${directory}/${options.name}/astro.config.mjs`)
      ).toBeTruthy();
      expect(
        tree.exists(`apps/${directory}/${options.name}/tsconfig.json`)
      ).toBeTruthy();
    });
  });

  describe('--tags', () => {
    test('should add project tags when provided', async () => {
      await applicationGenerator(tree, { ...options, tags: 'foo, bar' });

      const { tags } = readProjectConfiguration(tree, options.name);
      expect(tags).toEqual(['foo', 'bar']);
    });
  });

  describe('--integrations', () => {
    test('should add integrations to the astro config', async () => {
      (fetch as unknown as jest.Mock).mockImplementation((url) => {
        return Promise.resolve({
          json: () =>
            Promise.resolve(
              url.includes('@astrojs/solid-js')
                ? { name: '@astrojs/solid-js', version: '0.1.4' }
                : url.includes('@astrojs/vue')
                ? { name: '@astrojs/vue', version: '0.1.5' }
                : { name: 'should-not-be-installed' }
            ),
        });
      });

      await applicationGenerator(tree, {
        ...options,
        integrations: ['solid-js', 'vue'],
      });

      expect(
        tree.read(`${options.name}/astro.config.mjs`, 'utf-8')
      ).toMatchSnapshot();
    });

    test('should install provided integrations and dependencies', async () => {
      (fetch as unknown as jest.Mock).mockImplementation((url) => {
        return Promise.resolve({
          json: () =>
            Promise.resolve(
              url.includes('@astrojs/solid-js')
                ? {
                    name: '@astrojs/solid-js',
                    version: '0.1.4',
                    peerDependencies: { 'solid-js': '^1.3.6' },
                  }
                : url.includes('@astrojs/vue')
                ? {
                    name: '@astrojs/vue',
                    version: '0.1.5',
                    peerDependencies: { vue: '^3.2.30' },
                  }
                : { name: 'should-not-be-installed' }
            ),
        });
      });

      await applicationGenerator(tree, {
        ...options,
        integrations: ['solid-js', 'vue'],
      });

      const { devDependencies } = readJson(tree, 'package.json');
      expect(devDependencies['@astrojs/solid-js']).toBe('^0.1.4');
      expect(devDependencies['solid-js']).toBe('^1.3.6');
      expect(devDependencies['@astrojs/vue']).toBe('^0.1.5');
      expect(devDependencies['vue']).toBe('^3.2.30');
      expect(devDependencies['should-not-be-installed']).toBeUndefined();
    });

    test('should support extra integrations that are not listed scoped to @astrojs', async () => {
      (fetch as unknown as jest.Mock).mockImplementation((url) => {
        return Promise.resolve({
          json: () =>
            Promise.resolve(
              url.includes('@astrojs/netlify')
                ? { name: '@astrojs/netlify', version: '0.4.4' }
                : { name: 'should-not-be-installed' }
            ),
        });
      });

      await applicationGenerator(tree, {
        ...options,
        integrations: ['netlify'],
      });

      const { devDependencies } = readJson(tree, 'package.json');
      expect(devDependencies['@astrojs/netlify']).toBe('^0.4.4');
      expect(devDependencies['should-not-be-installed']).toBeUndefined();
    });

    test('should support third-party integrations with a name starting with "astro-" and with or without scope', async () => {
      (fetch as unknown as jest.Mock).mockImplementation((url) => {
        return Promise.resolve({
          json: () =>
            Promise.resolve(
              url.includes('astro-imagetools')
                ? { name: 'astro-imagetools', version: '0.6.10' }
                : url.includes('@lloydjatkinson/astro-snipcart')
                ? { name: '@lloydjatkinson/astro-snipcart', version: '0.1.2' }
                : { name: 'should-not-be-installed' }
            ),
        });
      });

      await applicationGenerator(tree, {
        ...options,
        integrations: ['astro-imagetools', '@lloydjatkinson/astro-snipcart'],
      });

      const { devDependencies } = readJson(tree, 'package.json');
      expect(devDependencies['astro-imagetools']).toBe('^0.6.10');
      expect(devDependencies['@lloydjatkinson/astro-snipcart']).toBe('^0.1.2');
      expect(devDependencies['should-not-be-installed']).toBeUndefined();
    });
  });

  describe('--addCypressTests', () => {
    test('should not add an e2e project when --addCypressTests=false', async () => {
      const e2eProjectName = `${options.name}-e2e`;

      await applicationGenerator(tree, { ...options, addCypressTests: false });

      expect(tree.exists(`${e2eProjectName}`)).toBeFalsy();
      const projects = getProjects(tree);
      expect(projects.has(e2eProjectName)).toBeFalsy();
    });

    test('should add an e2e project by default', async () => {
      const e2eProjectName = `${options.name}-e2e`;

      await applicationGenerator(tree, options);

      expect(tree.exists(`${e2eProjectName}`)).toBeTruthy();
      expect(readProjectConfiguration(tree, e2eProjectName)).toBeTruthy();
    });

    test('should add an e2e project when --addCypressTests=true', async () => {
      const e2eProjectName = `${options.name}-e2e`;

      await applicationGenerator(tree, { ...options, addCypressTests: true });

      expect(tree.exists(`${e2eProjectName}`)).toBeTruthy();
      expect(readProjectConfiguration(tree, e2eProjectName)).toBeTruthy();
    });

    test('should configure the e2e target correctly', async () => {
      const e2eProjectName = `${options.name}-e2e`;

      await applicationGenerator(tree, { ...options, addCypressTests: true });

      const e2eProject = readProjectConfiguration(tree, e2eProjectName);
      expect(e2eProject.targets.e2e).toBeTruthy();
      expect(e2eProject.targets.e2e.options.devServerTarget).toBe(
        `${options.name}:dev`
      );
      expect(e2eProject.targets.e2e.configurations).toBeUndefined();
    });

    test('should be configured correctly when passing a directory', async () => {
      const directory = `some-directory/sub-directory/${options.name}`;
      const e2eProjectName = `${options.name}-e2e`;

      await applicationGenerator(tree, {
        ...options,
        directory,
        addCypressTests: true,
      });

      expect(tree.exists(`${directory}-e2e`)).toBeTruthy();
      const e2eProject = readProjectConfiguration(tree, e2eProjectName);
      expect(e2eProject.targets.e2e).toBeTruthy();
    });
  });
});
