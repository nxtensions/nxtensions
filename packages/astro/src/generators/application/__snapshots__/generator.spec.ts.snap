// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application generator --integrations should add integrations to the astro config 1`] = `
"import { defineConfig } from \\"astro/config\\";
import solid from '@astrojs/solid-js';
import vue from '@astrojs/vue';

export default defineConfig({
  outDir: '../dist/app1',
  integrations: [solid(),vue()],
});
"
`;

exports[`application generator should add project configuration 1`] = `
Object {
  "$schema": "../node_modules/nx/schemas/project-schema.json",
  "name": "app1",
  "projectType": "application",
  "root": "app1",
  "sourceRoot": "app1/src",
  "tags": Array [],
  "targets": Object {
    "build": Object {
      "executor": "@nxtensions/astro:build",
      "options": Object {},
      "outputs": Array [
        "{workspaceRoot}/dist/{projectRoot}",
      ],
    },
    "check": Object {
      "executor": "@nxtensions/astro:check",
    },
    "dev": Object {
      "executor": "@nxtensions/astro:dev",
      "options": Object {},
    },
    "preview": Object {
      "dependsOn": Array [
        Object {
          "projects": "self",
          "target": "build",
        },
      ],
      "executor": "@nxtensions/astro:preview",
      "options": Object {},
    },
    "sync": Object {
      "executor": "@nxtensions/astro:sync",
    },
  },
}
`;
